pkg/vm/vm.go

```diff
@@ -236,12 +236,84 @@ func (v *VM) Step() {
 	}
 }
 
+// StepInto behaves the same as “step over” in case if the line does not contain a function it otherwise
+// the debugger will enter the called function and continue line-by-line debugging there.
+func (v *VM) StepInto() {
+	ctx := v.Context()
+
+	if ctx != nil {
+		v.state |= haltState
+	}
+
+	if v.HasStopped() {
+		return
+	}
+
+	if ctx != nil && ctx.prog != nil {
+		var op Instruction
+		if ctx.ip >= len(ctx.prog) {
+			op = RET
+		} else {
+			op = ctx.Next()
+		}
+		v.execute(ctx, op)
+		i, op := ctx.CurrInstr()
+		fmt.Printf("at breakpoint %d (%s)\n", i, op.String())
+	}
+
+	cctx := v.Context()
+	if cctx != nil && cctx.atBreakPoint() {
+		v.state = breakState
+	}
+}
+
+// StepOut takes the debugger to the line where the current function was called.
+func (v *VM) StepOut() {
+	if v.state == breakState {
+		v.state = noneState
+	} else {
+		v.state = breakState
+	}
+
+	expSize := v.istack.len
+	for v.state.HasFlag(noneState) && v.istack.len >= expSize {
+		v.StepInto()
+	}
+}
+
+// StepOver takes the debugger to the line that will step over a given line.
+// If the line contains a function the function will be executed and the result returned without debugging each line.
+func (v *VM) StepOver() {
+	if v.HasStopped() {
+		return
+	}
+
+	if v.state == breakState {
+		v.state = noneState
+	} else {
+		v.state = breakState
```

Comment:
Does it really make sense reassigning something that already is in the `v.state`?

Reply:
totally not.. heh =/